core/src/main/java/org/elasticsearch/cluster/routing/RoutingChangesObserver.java

```diff
@@ -120,6 +125,11 @@ public void startedPrimaryReinitialized(ShardRouting startedPrimaryShard, ShardR
         public void replicaPromoted(ShardRouting replicaShard) {
 
         }
+
+        @Override
+        public void initializedReplicaReinitialized(ShardRouting initializingReplica, ShardRouting reinitializedReplica) {
```

Comment:
1) There is an implementation of the interface (RoutingNodesChangedObserver) where I want to be 100% sure that every method that is added to the interface is properly implemented in that class. With default methods, it's easy to miss this.
2) I've renamed the method parameter.